src/search/landmarks/exploration.cc

```diff
@@ -147,124 +122,91 @@ void Exploration::build_unary_operators(const OperatorProxy &op) {
 */
 void Exploration::setup_exploration_queue(
     const State &state, const vector<FactPair> &excluded_props,
-    const unordered_set<int> &excluded_op_ids, bool use_h_max) {
+    const unordered_set<int> &excluded_op_ids) {
     prop_queue.clear();
 
     for (size_t var_id = 0; var_id < propositions.size(); ++var_id) {
         for (size_t value = 0; value < propositions[var_id].size(); ++value) {
             ExProposition &prop = propositions[var_id][value];
-            prop.h_add_cost = -1;
+            prop.excluded = false;
             prop.h_max_cost = -1;
-            prop.depth = -1;
-            prop.marked = false;
         }
     }
 
     for (const FactPair &fact : excluded_props) {
         ExProposition &prop = propositions[fact.var][fact.value];
-        prop.h_add_cost = -2;
+        prop.excluded = true;
     }
```

Comment:
I'm wondering when would be the best time to do this. I see it fitting in this issue rather than a follow up, but we wouldn't get rid of the loop over all propositions completeley (only the setting `excluded` to false part) since we need to reset the `h_max_cost`. I therefore consider just writing a comment for now, describing your suggestion and leaving it for when we work on issue1045 to replace the `h_max_cost` with a `reached` flag anyway. I believe the `reached` would still need a reset in every exploration, but maybe there's another smart way to avoid doing that for all propositions, I didn't think about it for too long.

Reply:
Sure, there may still be other O(|atoms|) processing here. But less processing will be faster. I have no strong preference regarding when to look into this, with a weak preference to merging soon provided we keep the momentum and continue working on this.

Reply:
I decided to address this right away and made the change within this issue. I've pushed the changes a minute ago, so you should be able to see the change now. This requires another experiment, so I'm setting this up now.